src/software/ai/hl/stp/play/shoot_or_pass_play.cpp

```diff
@@ -59,83 +59,184 @@ void ShootOrPassPlay::getNextTactics(TacticCoroutine::push_type &yield,
+                ->updateControlParams(world.ball().position(), CreaseDefenderAlignment::LEFT);
+        std::get<1>(crease_defender_tactics)
+                ->updateControlParams(world.ball().position(),
+                                      CreaseDefenderAlignment::RIGHT);
+
         auto pass1 = pass_eval.getBestPassInZones(cherry_pick_region_1).pass;
         auto pass2 = pass_eval.getBestPassInZones(cherry_pick_region_2).pass;
 
-        auto cherry_pick_tactic_1 = std::make_shared<MoveTactic>(false);
-        auto cherry_pick_tactic_2 = std::make_shared<MoveTactic>(false);
         cherry_pick_tactic_1->updateControlParams(pass1.receiverPoint(),
                                                   pass1.receiverOrientation(), 0.0,
                                                   MaxAllowedSpeedMode::PHYSICAL_LIMIT);
         cherry_pick_tactic_2->updateControlParams(pass2.receiverPoint(),
                                                   pass2.receiverOrientation(), 0.0,
-                                                  MaxAllowedSpeedMode::PHYSICAL_LIMIT);
-
-
-        do
-        {
-            attacker->updateControlParams(pass);
-            receiver->updateControlParams(pass);
-
-            std::get<0>(crease_defender_tactics)
-                ->updateControlParams(world.ball().position(),
-                                      CreaseDefenderAlignment::LEFT);
-            std::get<1>(crease_defender_tactics)
-                ->updateControlParams(world.ball().position(),
-                                      CreaseDefenderAlignment::RIGHT);
-            if (!attacker->done())
-            {
-                yield({{attacker, receiver},
-                       {cherry_pick_tactic_1, std::get<0>(crease_defender_tactics),
-                        std::get<1>(crease_defender_tactics)}});
-            }
-            else
-            {
-                yield({{receiver},
-                       {cherry_pick_tactic_1, cherry_pick_tactic_2},
-                       {std::get<0>(crease_defender_tactics),
-                        std::get<1>(crease_defender_tactics)}});
-            }
-        } while (!receiver->done());
-    }
-    else
-    {
-        LOG(DEBUG) << "Took shot";
-    }
-
-    LOG(DEBUG) << "Finished";
+                                                  MaxAllowedSpeedMode ::PHYSICAL_LIMIT);
+
+        auto min_score_to_commit_to_pass = play_config->getShootOrPassPlayConfig()->getAbsMinPassScore()->value();
+        if(current_committed_pass.has_value()) {
+            current_committed_pass->rating = ratePass(world, current_committed_pass->pass, world.field().fieldLines(), play_config->getPassingConfig());
+        }
+        if (!current_committed_pass.has_value() && best_pass_and_score_so_far.rating > min_score_to_commit_to_pass) {
+            current_committed_pass = best_pass_and_score_so_far;
+        }else if(!attacker->done() && current_committed_pass.has_value() && current_committed_pass->rating < 0.01) {
```

Comment:
We only want to abort the pass **if we haven't already kicked it**. We don't want to abort if the ball is in motion and the receiver is in the middle of receiving.